File: docs/snippet/code-signal/arcade/intro/the-journey-begins/check-palindrome/check-palindrome.py
# def check_palindrome(input_string):
#     return input_string == input_string[::-1]

def check_palindrome(input_string):
    start = 0
    middle = len(input_string) // 2
    end = len(input_string) - 1

    while start <= middle:
        if input_string[start] is not input_string[end]:
            return False
        start = start + 1
        end = end - 1

    return True


print(check_palindrome('saippuakivikauppias'))

File: docs/snippet/code-signal/arcade/intro/the-journey-begins/check-palindrome/check-palindrome.cs
using Internal;

// static bool checkPalindrome(string inputString)
// {
//   return inputString == string.Join("", inputString.end());
// }

static bool checkPalindrome(string inputString)
{
  int start = 0;
  int middle = inputString.Length / 2;
  int end = inputString.Length - 1;

  while (start <= middle)
  {
    if (inputString[start] != inputString[end])
    {
      return false;
    }
    start = start + 1;
    end = end - 1;
  }

  return true;
}


Console.WriteLine($"{checkPalindrome("saippuakivikauppias")}");
